src/main/java/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,23 @@
+package service.impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(transaction);
```

You call `operationStrategy.getHandler(transaction.getOperation())` and immediately call `handler.apply(transaction)` without checking for null. OperationStrategy.getHandler currently can return null when a handler is not registered; the task requires the strategy to throw or otherwise fail fast when a handler is missing ("OperationStrategy holds a map ... throw if handler missing"). Either ensure OperationStrategyImpl throws when handler is absent or check `handler==null` here and throw an informative RuntimeException such as "No handler for operation: X". See OperationStrategyImpl and the requirement for handler resolution behavior  .